File: remove_element_ll/rm_el_ll.py
# Definition for singly-linked list.
class ListNode:
  def __init__(self, val=0, next=None):
    self.val = val
    self.next = next

  def print(self):
    curr = self
    while curr:
      print(curr.val, end='')
      if curr.next:
        print("->", end='')
      curr = curr.next
    print()

def buildLinkedList(elements: dict):
  if len(elements) == 0: return None
  head = ListNode(elements[0])
  curr = head
  for i in range(1, len(elements)):
    curr.next = ListNode(elements[i])
    curr = curr.next
  return head


# solution
def removeElements(head: ListNode, val: int) -> ListNode:
  return head


# head = buildLinkedList([1,2,6,3,4,5,6]); val = 6; # 1->2->3->4->5
# head = buildLinkedList([1,1,2,6,3,1,4,5,6]); val = 1; # 2,6,3,4,5,6
head = buildLinkedList([1,2]); val = 2; # 1

res = removeElements(head, val)

if res: res.print()

File: remove_element_ll/rm_el_ll.rb
# Definition for singly-linked list.
class ListNode
  attr_accessor :val, :next
  def initialize(val = 0, _next = nil)
    @val = val
    @next = _next
  end

  def to_s
    cur = self
    res = []
    while cur
      res << cur.val
      res << '->' if cur.next
      cur = cur.next
    end
    res.join
  end
end

def build_linked_list(arr)
  return nil if arr.size.zero?
  cur = head = ListNode.new(arr[0])
  for i in 1...arr.size
    cur.next = ListNode.new(arr[i])
    cur = cur.next
  end
  head
end

# @param {ListNode} head
# @param {Integer} val
# @return {ListNode}
def remove_elements(head, val)
  return head unless head
  #
  return head
end


# head = build_linked_list([1,2,6,3,4,5,6]); val = 6; # 1->2->3->4->5
head = build_linked_list([1,1,2,6,3,1,4,5,6]); val = 1; # 2,6,3,4,5,6
# head = build_linked_list([1,2]); val = 2; # 1
# head = build_linked_list([]); val = 0

res = remove_elements(head, val)

puts res.to_s
